### ISSUE TITLE
optimize parallel downloads 

### ISSUE THREAD

**Issue by jakoch** [WPN-XM/WPN-XM#437]
It would be nice to further optimize the parallel downloads during the build.
I've already reduced the downloads of the installer main packages.
The main packages are all "Full installers" and the "LiteRC" installers.
The downloads of all others installers are derived from the "Full Installation" download folders.
But unfortunately there are still many packages with unnecessary duplicate downloads.

The next step is to merge the download description files into one file
and download each component only once. 
This will reduce the number of downloads, used bandwidth and build time.

----

- merge the download description files for aria into one array
- apply array_unique() to drop duplicate downloads
- then write the main download file, called "pool.txt" with target folder "pool"
- execute aria to process this file and place the downloads into "pool"
- process each download description file, except "pool.txt" 
   - get component name 
   - lookup archive name in main download description file
   - copy the archive into the individual folders for this installer

----

Blocked: 
We need to resolve the download filename problem first
The download description files for aria contains only a simply archive name without bitsize
- the bitsize has to be added to the downloaded archives, in order to avoid name clashes
 - e.g. its not possible to have `imagick.zip` for x86 and x64
   - it needs to become `imagick.zip` and `imagick-x64.zip`
- This needs a change in the main download description file, too.
- Alternative: two folders x86 / x64 could also work out.. not sure.

**Comment by jakoch** [comment]
The downloader hits the memory limit on Travis.
This feature is needed to get the next release out.

**Comment by jakoch** [comment]
1. Initially we downloaded each component for each installer. 20 installers means 20 downloads for each components. This took ages and caused a lot of traffic. Fetching on every matrix slot is still the native way, when working with a build matrix without a shared cache between build instances. 

2. Then we reduced the number of downloads by downloading only the components of each full installer (6) and populate the other installers (Lite, Standard) from there. 
Still, six downloads for MariaDb (three downloads for x86 and three for x64).
4 downloads too much for components with x86/x64 packaging.

3. Finally, we download each component once into the shared folders for the two bitsizes (x86/x64)
and populate all installer download folders from there. Except for PHP and PHP Extensions, which are  downloaded directly into the download folder for the Full installer and all other installers are populated from there. :sunglasses:
